### PR TITLE
Fix mouse and multitouch controls when 3D cameras are rotated

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: GDJS/Runtime/pixi-renderers/layer-pixi-renderer.ts [hunk 1]
```diff
@@ -343,6 +345,47 @@ namespace gdjs {
       }
     }
 
+    transformToWorld(
+      screenX: float,
+      screenY: float,
+      worldZ: float,
+      cameraId: integer,
+      result: FloatPoint
+    ) {
```
Comment (AlexandreSi) [inline GDJS/Runtime/pixi-renderers/layer-pixi-renderer.ts:354]: Can you type the return value? `FloatPoint`

File: GDJS/Runtime/pixi-renderers/layer-pixi-renderer.ts [hunk 2]
```diff
@@ -343,6 +345,55 @@ namespace gdjs {
...
 
+    isCameraRotatedIn3D() {
+      return (
+        this._threeCamera &&
+        this._threeCamera.rotation.x !== 0 &&
+        this._threeCamera.rotation.y !== 0
+      );
+    }
+
+    transformToWorld(
```
Comment (4ian) [inline GDJS/Runtime/pixi-renderers/layer-pixi-renderer.ts:356]: transformTo3DWorld?

File: GDJS/Runtime/layer.ts [hunk 3]
```diff
@@ -238,6 +240,11 @@ namespace gdjs {
 
       // The result parameter used to be optional.
       let position = result || [0, 0];
+
+      if (this._renderer.isCameraRotatedIn3D()) {
```
Comment (4ian) [inline GDJS/Runtime/layer.ts:244]: I'm a bit sad about this thing that mix the responsibility between the renderer and the layer.ts file. Maybe we can move the convertCoords at some point (and the other methods) to the renderer. Or keep all the logic here?

**Review by 4ian — APPROVED** [review]

File: GDJS/Runtime/layer.ts (hunk 3 above)
Comment (4ian) [reply GDJS/Runtime/layer.ts:244]: Add a comment "TODO (3D): this should in theory be in a single file".

Comment (D8H) [reply GDJS/Runtime/layer.ts:244]: Yes, there is probably a decision to make.
Should GDJS handle 3D transformations itself?
We did for 2D, but calculus are more complicated in 3D. It's probably something to leave to Three.js.

**Pull request merged**